src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/CreateReindexRequestHandler.cs

```diff
@@ -69,16 +73,62 @@ public async Task<CreateReindexResponse> Handle(CreateReindexRequest request, Ca
             // not detect the resources that need to be reindexed.
             await _searchParameterOperations.GetAndApplySearchParameterUpdates(cancellationToken);
 
+            // What this handles is the scenario where a user is effectively forcing a reindex to run
+            // by passing in a parameter of targetSearchParameterTypes. From those we can identify the base
+            // resource types. With the resource types we can now identify from our SearchParameterHashMap
+            // any matches and reset the hash value with a new guid thereby kicking off a new reindex job.
+            // Of course if we don't find a match then we leave the hash value unaffected and pass it through.
+            Dictionary<string, string> hashMap = null;
+            var searchParameterResourceTypes = new HashSet<string>();
+            if (request.TargetSearchParameterTypes.Any())
+            {
+                foreach (var searchParameterUrl in request.TargetSearchParameterTypes)
+                {
+                    SearchParameterInfo searchParameterInfo = _searchParameterDefinitionManager.GetSearchParameter(searchParameterUrl);
+                    if (searchParameterInfo == null)
+                    {
+                        throw new RequestNotValidException(Core.Resources.SearchParameterByDefinitionUriNotSupported, searchParameterUrl);
+                    }
+
+                    if (searchParameterInfo != null)
```

Comment:
Do we need this check?  We checked for == null above, so at this point it seems that searchParameterInfo must be other than null.

Reply:
Yep, thanks for finding that.